c10/core/inference_mode.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
```

Comment:
bikeshed: can we name this file after the name of the main class it exports, in the same way the other files in this directory are named? :P